README.md

```diff
@@ -8,98 +8,71 @@ title: Spark on Kubernetes Integration Tests
-phase must be run every time the Spark main code changes.  When running tests
-from the command line, the `pre-integration-test` phase should automatically be
-invoked if the `integration-test` phase is run.
-
-With Maven, the integration test can be run using the following command:
-
-```
-$ mvn clean integration-test  \
-    -Dspark-distro-tgz=spark/spark-2.3.0-SNAPSHOT-bin.tgz
-```
-
-## Running against an arbitrary cluster
-
-In order to run against any cluster, use the following:
-```sh
-$ mvn clean integration-test  \
-    -Dspark-distro-tgz=spark/spark-2.3.0-SNAPSHOT-bin.tgz  \
-    -DextraScalaTestArgs="-Dspark.kubernetes.test.master=k8s://https://<master>
-
-## Reuse the previous Docker images
-
-The integration tests build a number of Docker images, which takes some time.
-By default, the images are built every time the tests run.  You may want to skip
-re-building those images during development, if the distribution package did not
-change since the last run. You can pass the property
-`spark.kubernetes.test.imageDockerTag` to the test process and specify the Docker 
-image tag that is appropriate.
-Here is an example:
-
-```
-$ mvn clean integration-test  \
-    -Dspark-distro-tgz=spark/spark-2.3.0-SNAPSHOT-bin.tgz  \
-    -Dspark.kubernetes.test.imageDockerTag=latest
-```
+The simplest way to run the integration tests is to install and run Minikube, then run the following:
+
+    build/mvn integration-test
+    
+The minimum tested version of Minikube is 0.23.0. The kube-dns addon must be enabled. Minikube should
+run with a minimum of 3 CPUs and 4G of memory:
+
+    minikube start --cpus 3 --memory 4G
+
+You can download Minikube [here](https://github.com/kubernetes/minikube/releases).
+
+# Integration test customization
+
+Configuration of the integration test runtime is done through passing different Java system properties to the Maven
+command. The main useful options are outlined below.
+
+## Use a non-local cluster
+
+To use your own cluster running in the cloud, set the following:
+
+* `spark.kubernetes.test.deployMode` to `cloud` to indicate that Minikube will not be used.
+* `spark.kubernetes.test.master` to your cluster's externally accessible URL
+* `spark.kubernetes.test.imageRepo` to a write-accessible Docker image repository that provides the images for your
+cluster. The framework assumes your local Docker client can push to this repository.
+
+Therefore the command looks like this:
+
+    build/mvn integration-test \
+      -Dspark.kubernetes.test.deployMode=cloud \
+      -Dspark.kubernetes.test.master=https://example.com:8443/apiserver \
+      -Dspark.kubernetes.test.repo=docker.example.com/spark-images
+      
+## Re-using Docker Images
+
+By default, the test framework will build new Docker images on every test execution. A unique image tag is generated,
+and it is written to file at `target/imageTag.txt`. To reuse the images built in a previous run, set:
```

Comment:
I liked before that it was using the git sha - it made it easy to correlate to the distribution that was just tested.

Reply:
Discussed in https://github.com/apache-spark-on-k8s/spark-integration/pull/31#discussion_r161349039